### PR TITLE
formatted README, added content

### PR DESCRIPTION
**Pull request by kw-fn**

### CODE REVIEW

**Comment by kw-fn** [comment]
I forgot about some changes and wasn't expecting merge conflicts.., I can flatten this PR if you'd like @glyn

**Comment by kw-fn** [comment]
Oh oops I didn't realize you could squash on GitHub, hold on let me try to revert this so it only flattens not merges.

**Comment by kw-fn** [comment]
Maybe it's better to just leave this as it is..

File: src/applications.md [hunk 1]
```diff
@@ -24,5 +25,7 @@ Applications of Rust:
 * [_nalgebra_](https://nalgebra.org/docs/) - a linear algebra library
 * [Prusti user guide](https://viperproject.github.io/prusti-dev/user-guide/) - a Rust verifier
 * [Serde](https://serde.rs/) - **ser**ialize and **de**serialize Rust data structures
+* [_Solana_](https://docs.solana.com/) - web-scale blockchain for decentralized apps
+* [_tantivy_](https://tantivy-search.github.io/examples/basic_search.html) - a full text search engine library
 * [Tealdeer User Manual](https://dbrgn.github.io/tealdeer/) - implementation of [tldr](https://github.com/tldr-pages/tldr) (help like `man`, but more approachable).
```
Comment (glyn) [inline src/applications.md:30]: @kw-fn I don't think the above two should be included because they don't even look slightly like they are in mdbook format. My rationale for restricting to "books" is that this is the Little Book of Rust _Books_. If we drop that restriction, we'll end up listing hundreds of Rust web sites and this book will lose its attraction.

File: README.md [hunk 2]
```diff
@@ -34,18 +34,14 @@ change.
...
-Each entry is of the form:
-```
-* [The Literal Title of The Book](https://a-link-to-the/book/) - an optional comment if the title isn't self-explanatory
-```
-
-The `.editorconfig` file captures basic source files formatting conventions.
-Many editors support this file natively; others (such as VS code) require a plugin, see https://editorconfig.org/.
-
-The default branch is `main`. See [Renaming the default branch from master](https://github.com/github/renaming) for the rationale.
+Basic conventions around source files formatting are captured in the
```
Comment (glyn) [inline README.md:37]: @kw-fn Why did you delete the entry format convention? I think PR authors need to know the intended format so they can adhere to it. Otherwise they have to guess. Please could you put that back in?

**Comment by glyn** [comment]
@kw-fn I went ahead and reverted the parts of this PR that I didn't want to include.

File: README.md [hunk 3]
```diff
@@ -34,18 +34,14 @@ change.
...
+`.editorconfig` file. Many editors support that file natively. Others (such as
+VS code) require a plugin, see https://editorconfig.org/.
 
-Commit logs should describe the specific changes made by a commit. See [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
```
Comment (glyn) [inline README.md:47]: @kw-fn I want to keep in the guidance about commit logs.

File: README.md (hunk 2 above)
Comment (glyn) [reply README.md:37]: I put it back in.

File: src/applications.md (hunk 1 above)
Comment (glyn) [reply src/applications.md:30]: I removed these two.

**Review by glyn — COMMENTED** [review]
I reverted the changes I didn't like. Please see the comments here for my reasoning.

File: src/applications.md (hunk 1 above)
Comment (kw-fn) [reply src/applications.md:30]: Actually I think these are both special cases from a couple points-of-view, I can provide some context for you:

1. They can be seen reference implementations of the services they provide, and,
2. they use novel and sophisticated approaches to documentation.

It's probably more arguable that adding `mdbook` formatted content would lead to bloat since that is the default being used nearly everywhere. Here the consideration was the novelty of the custom documentation and the relative importance of each crate within its ecosystem.

File: README.md (hunk 2 above)
Comment (kw-fn) [reply README.md:37]: There is only a single branch to the project as far as I can determine. Also I believe the use of a plugin should not necessarily be a recommendation given that simple editing oversight can accomplish what the dot-file seeks to enforce.

File: README.md (hunk 3 above)
Comment (kw-fn) [reply README.md:47]: I'm not sure I see the need for enforcing this, or making possible contributors feel they need to log edits to documentation.

Comment (glyn) [reply README.md:47]: It's just a general hygiene factor for git repositories, for documentation or code. I'd prefer to keep it.

File: README.md (hunk 2 above)
Comment (glyn) [reply README.md:37]: Yes, there's a single branch, but the adoption of `main` as the default branch is a relatively recent convention which I wanted to explain.

The editor config file is a general hygiene item. It's not that important, but I'd like to keep it because I want to promote its use in other repos.

File: src/applications.md (hunk 1 above)
Comment (glyn) [reply src/applications.md:30]: > Actually I think these are both special cases from a couple points-of-view, I can provide some context for you:
> 
>     1. They can be seen reference implementations of the services they provide, and,
> 
>     2. they use novel and sophisticated approaches to documentation.
> 
> 

I had another look at these two. I agree that Solana is close to book format and I'd be happy to have it included. But the tantivy site isn't anything like a book really and I feel it would dilute the LBORB concept of being a collection of _books_.

> It's probably more arguable that adding `mdbook` formatted content would lead to bloat since that is the default being used nearly everywhere. Here the consideration was the novelty of the custom documentation and the relative importance of each crate within its ecosystem.

The concept of LBORB, which got hundreds of positive upvotes on [Reddit](https://www.reddit.com/r/rust/comments/kwiwb8/the_little_book_of_rust_books/), was to collect together mdbook format books. I'd like to maintain that concept and I hope we can align on that.

Comment (kw-fn) [reply src/applications.md:30]: In some sense I can see there is value in denoting which links don't follow along with a certain predominant template. On the other hand I do not see any utility in excluding content solely based on it not conforming exactly enough to some novel criteria. 

As an aside here I do think that tantivy is formatted in a way that does meet the criteria of being a book, which I think is to say that it encodes logical separation based on chapters, sections, and sub-sections.

Thanks for your efforts in maintaining the standards.

Comment (glyn) [reply src/applications.md:30]: > In some sense I can see there is value in denoting which links don't follow along with a certain predominant template. On the other hand I do not see any utility in excluding content solely based on it not conforming exactly enough to some novel criteria.

I think we'll have to agree to differ on this point.

> 
> As an aside here I do think that tantivy is formatted in a way that does meet the criteria of being a book, which I think is to say that it encodes logical separation based on chapters, sections, and sub-sections.

That is true, but I would say the books in LBORB generally try quite hard to be self-contained and define their terms, another facet of a book in addition to its structure into chapters etc.. The tantivy document is really just an example which assumes quite a bit of prior understanding of tantivy (such as what the name even means), so I don't think it falls into the "hono[u]rable mention" category of non-mdbook books which deserve including because of their exceptional value.

> 
> Thanks for your efforts in maintaining the standards.

You're welcome. It has been helpful for you to challenge the standards a little as it helps to clarify the criteria for inclusion.
